### PR TITLE
Ryan dev3

### PR DESCRIPTION
**Pull request by ryanwooa**
- update to new Application structure, migrate components to new structure
- fix setting router problem
- part finish 404 page

### CODE REVIEW

File: components/application/widgets/Modal.js
Comment (jlmjkfd) [inline components/application/widgets/Modal.js:1]: Cannot dismiss the modal in my browser.
When I click the overlay (call the onDismiss function), the modal won't be dismissed. If I click the back button of the browser, the page goes back to the one without the modal, then all the router links will be dead, including the setting link and the links on the left bar.

File: app/application/@settingModal/(.)setting/[settingItem]/page.js
Comment (jlmjkfd) [inline app/application/@settingModal/(.)setting/[settingItem]/page.js:1]: What does {" "} do in the link 'theme'?

File: styles/scss/application.module.scss
Comment (jlmjkfd) [inline styles/scss/application.module.scss:1]: We don't need a unit after 0 value. (width: 0px; --> width: 0;)

File: styles/scss/leftbar.module.scss
Comment (jlmjkfd) [inline styles/scss/leftbar.module.scss:1]: 0px --> 0

File: styles/scss/todoList.module.scss
Comment (jlmjkfd) [inline styles/scss/todoList.module.scss:1]: 0px --> 0

File: app/application/@settingModal/(.)setting/[settingItem]/page.js
Comment (ryanwooa) [reply app/application/@settingModal/(.)setting/[settingItem]/page.js:1]: some random code generate by VScode. I delete it

File: components/application/widgets/Modal.js
Comment (ryanwooa) [reply components/application/widgets/Modal.js:1]: I can dismiss Modals in two browers's, Edge and Vivaldi, all based on Chrome.  Try refresh, then test? Or update packages?

Comment (jlmjkfd) [reply components/application/widgets/Modal.js:1]: Refreshing it doesn't work. My Chrome version is 119.0.6045.160. Which packages need to be updated?

Comment (ryanwooa) [reply components/application/widgets/Modal.js:1]: My Edge is 119.0.2151.72. Have your Next updated to 14? I dont know other possibility

Comment (jlmjkfd) [reply components/application/widgets/Modal.js:1]: Yes, 14.0.2. We may need other collaborators to test.

Comment (zli213) [reply components/application/widgets/Modal.js:1]: It works in my browsers. I'm using Arc and the Chrome version is 119.0.6045.159.

**Review by zli213 — COMMENTED** [review]
No problems.

**Pull request merged**